### ISSUE TITLE
Does your training method apply intermediate supervision?

### ISSUE THREAD

**Issue by Gavinwxy** [1adrianb/face-alignment#74]
Hi, I am a newcomer of face alignment work and trying to train your model on my own data set. I read about hourglass model and find it is trained by intermediate supervision in previous work. I read your torch 7 training code (I am actually not familiar with torch 7) and still can not find the answer. Could your please tell me if you compute loss for each hourglass stage and do backprop or just for the final stage. Thanks to your work!

**Comment by 1adrianb** [comment]
Hi @Gavinwxy, yes the model is trained with intermediate supervision. In this case there is a stack of 4 so the loss is aplied afeter every hg module.

**Comment by Gavinwxy** [comment]
@1adrianb If I understand right, for each hourglass stage there will be 68 heat maps generated and then compared with ground truth (also 68 heat maps drawn from annotation). Loss are calculated between heat maps and accumulated through the whole network. Only one backprop is applied for the final loss. Is this understanding correct? If not, how do you handle loss from different stages? Do you do backprop for each stage separately for one batch input?

**Comment by 1adrianb** [comment]
Yes, only one single backprop, where the loss is a sum of all losses (4 in this case).

**Comment by sunjunlishi** [comment]
Your effect is very good, just do not know how you train.
